run.py

```diff
@@ -1,25 +1,132 @@
 import os
 import sys
-from mig_dx_api import DX
+import google.auth
+from mig_dx_api import (
+    DX,
+    DatasetSchema,
+    Installation,
+    SchemaProperty
+)
+from mig_dx_api._dataset import DatasetOperations
+from google.cloud import bigquery
 
+# Pass in name of BigQuery dataset from ScriptRunner
+dataset_id = sys.argv[1]
+table_name = sys.argv[2]
+# TODO: Error if params are missing
+print(f"dataset_id {dataset_id} and table_name {table_name}")
+
+# Initialize the mig client
 private_key = f"-----BEGIN PRIVATE KEY-----\n{os.environ.get("PRIVATE_KEY")}\n-----END PRIVATE KEY-----"
-# Initialize the client
 dx = DX(app_id=os.environ.get("APP_ID"), private_key=private_key)
 dx.base_url = os.environ.get("BASE_URL")
-
 user_info = dx.whoami()
 print(f"user info: {user_info}\n")
 
-# Pass in name of BigQuery dataset from ScriptRunner
-dataset_id = sys.argv[1]
-table_name = sys.argv[2]
+# Initialize the google bigquery client
+credentials, project = google.auth.default(
+    scopes=[
+        "https://www.googleapis.com/auth/bigquery",
+    ]
+)
+print(f"bigquery project: {project}")
+client = bigquery.Client(credentials=credentials, project=project)
 
-print(f"dataset_id {dataset_id} and table_name {table_name}")
+def get_schema() -> DatasetSchema:
+    """
+    Get schema of source dataset from Portal BigQuery
+    """
+    dataset_ref = client.dataset(dataset_id=dataset_id, project=project)
+
+    table_ref = dataset_ref.table(table_name)
+
+    table = client.get_table(table_ref)
+
+    schema = table.schema
+    table_constraints = table.table_constraints # might not exist
+    primary_key = table_constraints.primary_key.columns if table_constraints else []
+
+    print(schema)
+    print(f"table constraints: {primary_key}")
+
+    properties = []
+    for field in schema:
+        # We don't care about the actual type of the field, the type is always "string"
+        property = SchemaProperty(name=field.name, type="string", required=not field.is_nullable)
+        properties.append(property)
+
+    return DatasetSchema(
+        properties=properties,
+        primary_key=primary_key
+    )
+
+def create_dataset(installation: Installation) -> DatasetOperations:
+    """
+    Create MiG dataset using mig-dx-api client
+    """
+    # Get schema of source dataset
+    dataset_schema = get_schema()
+
+    # create MiG dataset with source schema
+    with dx.installation(installation) as ctx:
+        dataset_name = f'{table_name}-phoenix'
```

Comment:
If we are creating a new dataset, should it just be named after the biquery table that we're copying from?